frontends/PyCDE/src/pycde/system.py

```diff
@@ -101,11 +101,24 @@ def add_packaging_step(self, func: Callable):
 
   @property
   def hw_output_dir(self):
-    return self.output_directory / "hw"
+    ret = self.output_directory / "hw"
+    if not ret.exists():
+      ret.mkdir()
+    return ret
 
   @property
   def runtime_output_dir(self):
-    return self.output_directory / "runtime"
+    ret = self.output_directory / "runtime"
+    if not ret.exists():
+      ret.mkdir()
+    return ret
+
+  @property
+  def sys_runtime_output_dir(self):
+    ret = self.runtime_output_dir / self.name
```

Comment:
nit: this pattern is repeated 3 times for 3 different folder names. private internal function with string argument would standardize.